frontend/js/src/explore/artist-similarity/Data.tsx

```diff
@@ -0,0 +1,244 @@
+  const COLOR_MIX_WEIGHT = 0.3;
+  // Score in case it is undefined (as in case of main artist)
+  const NULL_SCORE = Infinity;
+
+  const color1 = colorGenerator();
+  const color2 = color1.clone().tetrad()[1];
+  const [similarArtistsList, setSimilarArtistsList] = useState<
+    Array<ArtistType>
+  >([]);
+  // State to store the complete list of similar artists to help with when user changes artist limit
+  const [completeSimilarArtistsList, setCompleteSimilarArtistsList] = useState<
+    Array<ArtistType>
+  >([]);
+  const [mainArtist, setMainArtist] = useState<ArtistType>();
+  const [similarArtistsLimit, setSimilarArtistsLimit] = useState(
+    SIMILAR_ARTISTS_LIMIT_VALUE
+  );
+  const [colors, setColors] = useState([color1, color2]);
+  const [artistMBID, setArtistMBID] = useState(ARTIST_MBID);
+
+  const scoreList: Array<number> = [];
+
+  const processData = useCallback(
+    (dataResponse: ApiResponseType): void => {
+      // Type guard for dataset response
+      const isDatasetResponse = (
+        response: MarkupResponseType | DatasetResponseType
+      ): response is DatasetResponseType => {
+        return response.type === "dataset";
+      };
+      // Get the datasets out of the API response
+      const artistsData = dataResponse.filter(isDatasetResponse);
+      if (artistsData.length) {
+        // Get the main artist from the first dataset
+        setMainArtist(artistsData[0].data[0]);
+        // Get the similar artists from the second dataset
+        const similarArtistsResponse = artistsData[1];
+        if (similarArtistsResponse?.data?.length) {
+          setCompleteSimilarArtistsList(similarArtistsResponse.data);
+          setSimilarArtistsList(
+            similarArtistsResponse.data.slice(0, similarArtistsLimit)
+          );
+        }
+        // In case no similar artists are found
+        else {
+          setSimilarArtistsList([]);
+        }
+      }
+      setColors([tinycolor.mix(color1, color2, COLOR_MIX_WEIGHT), color2]);
+    },
+    [artistMBID]
+  );
+
+  const fetchData = useCallback(
+    async (artist_mbid: string): Promise<void> => {
+      try {
+        const response = await fetch(BASE_URL + artist_mbid);
+        const data = await response.json();
+        processData(data);
+      } catch (error) {
+        // Error message goes here.
+        alert(
+          "Something went wrong while loading information, please try again"
+        );
+      }
+    },
+    [processData]
+  );
+
+  // Update the graph when either artistMBID or similarArtistsLimit changes
+  useEffect(() => {
+    fetchData(artistMBID);
+  }, [artistMBID, fetchData]);
+  // Update the graph when limit changes by only changing the data and not making a new request to server
+  useEffect(() => {
+    const newSimilarArtistsList = completeSimilarArtistsList.slice(
+      0,
+      similarArtistsLimit
+    );
+    setSimilarArtistsList(newSimilarArtistsList);
+  }, [similarArtistsLimit]);
```

Comment:
The dependency list needs to be updated.

Reply:
Fixed